pkg/storage/queue.go

```diff
@@ -323,7 +323,7 @@ func (bq *baseQueue) MaybeAdd(repl *Replica, now hlc.Timestamp) {
 	bq.mu.Lock()
 	defer bq.mu.Unlock()
 
-	if bq.mu.stopped {
+	if bq.mu.stopped && bq.mu.disabled {
```

Comment:
If this line had been in place, then #12224 would have been sufficient as it would have prevented the calls into shouldQueue which panicked. However, I decided to make the rest of this change (essentially undoing #12224 in favor of a different solution) because it seems more robust to fix consistencyQueue to not panic than to require some set of tests to disable the consistency queue (were the tests affected by #12224 the only ones, or were there other tests that were at risk but for whatever reasons hadn't manifested the panic?)